util/http/http_client.cc

```diff
@@ -97,5 +108,47 @@ bool Client::IsConnected() const {
   return socket_ && socket_->IsOpen();
 }
 
+///////////////////////////////////////////////////////////////////////////////
+
+SSL_CTX* SecureClient::CreateSslContext(SecurityLevel sec_level) {
+  SSL_CTX* ctx = SSL_CTX_new(TLS_client_method());
+  if (ctx) {
+    long tls_ver = sec_level == LEVEL_5 ? TLS1_3_VERSION : TLS1_2_VERSION;
+    SSL_CTX_set_min_proto_version(ctx, tls_ver);
+    if (SSL_CTX_set_default_verify_paths(ctx) != 1) {
+      LOG(WARNING) << "failed to set default verify path on client context for TLS connection";
+    }
+    SSL_CTX_set_options(ctx, SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS);
+    SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, VerifyCallback);
+    SSL_CTX_set_verify_depth(ctx, 4);
+    // Can also be defined using "ADH:@SECLEVEL=0" cipher string below.
+    SSL_CTX_set_security_level(ctx, int(sec_level));
+    SSL_CTX_dane_enable(ctx);  // see https://www.internetsociety.org/resources/deploy360/dane/
+    DCHECK(1 == SSL_CTX_set_cipher_list(ctx, "ADH"));
```

Comment:
why ADH is important here? specifically it's confusing because this call won't happen in the release mode

Reply:
I will remove it